client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/RollupDocumentationIT.java

```diff
@@ -160,4 +162,61 @@ public void onFailure(Exception e) {
             assertTrue(latch.await(30L, TimeUnit.SECONDS));
         }
     }
+
+    public void testDeleteRollupJobAsync() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+
+        String id = "job_2";
+        // tag::x-pack-rollup-delete-rollup-job-request
+        DeleteRollupJobRequest request = new DeleteRollupJobRequest(id);
+        // end::x-pack-rollup-delete-rollup-job-request
+
+        // tag::x-pack-rollup-delete-rollup-job-execute-listener
+        ActionListener<DeleteRollupJobResponse> listener = new ActionListener<DeleteRollupJobResponse>() {
+            @Override
+            public void onResponse(DeleteRollupJobResponse response) {
+                boolean acknowledged = response.isAcknowledged(); // <1>
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                // <2>
+            }
+        };
+        // end::x-pack-rollup-delete-rollup-job-execute-listener
+
+        // Replace the empty listener by a blocking listener in test
+        final CountDownLatch latch = new CountDownLatch(1);
+        listener = new LatchedActionListener<>(listener, latch);
+
+        // tag::x-pack-rollup-delete-rollup-job-execute
+        //client.rollup().deleteRollupJob(request, RequestOptions.DEFAULT); // <1>
+        // end::x-pack-rollup-delete-rollup-job-execute
+
+        // tag::x-pack-rollup-delete-rollup-job-execute-async
+        client.rollup().deleteRollupJobAsync(request, RequestOptions.DEFAULT, listener); // <1>
+        // end::x-pack-rollup-delete-rollup-job-execute-async
+
+        assertTrue(latch.await(30L, TimeUnit.SECONDS));
+
+    }
+
+    public void testDeleteRollupJob() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+
+        String id = "job_2";
+        // tag::x-pack-rollup-delete-rollup-job-request
+        DeleteRollupJobRequest request = new DeleteRollupJobRequest(id);
+        // end::x-pack-rollup-delete-rollup-job-request
+
+        try {
+            // tag::x-pack-rollup-delete-rollup-job-execute
+            client.rollup().deleteRollupJob(request, RequestOptions.DEFAULT); // <1>
+            // end::x-pack-rollup-delete-rollup-job-execute
+        } catch (Exception e) {
+            // Swallow any exception, this test does not test actually cancelling.
```

Comment:
do we expect the deletion of a non existing job to fail? If so, can we also test the exception?

Reply:
We could, but is the DocumentationIT really the right place for this? I'll add to `RollupIT` as that seems a better location for this test.

Reply:
Right - RollupIT is the right place